Lib/fractions.py

```diff
@@ -313,7 +313,19 @@ def from_decimal(cls, dec):
             raise TypeError(
                 "%s.from_decimal() only takes Decimals, not %r (%s)" %
                 (cls.__name__, dec, type(dec).__name__))
-        return cls(*dec.as_integer_ratio())
+        return cls._from_pair(*dec.as_integer_ratio())
+
+    @classmethod
+    def _from_pair(cls, numerator, denominator, /):
```

Comment:
Can we rename `_from_pair` to something more descriptive? How about `_from_coprime_pair` or `_from_coprime_ints`?